util/EzyRoundRobin.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using com.tvd12.ezyfoxserver.client.function;
+
+namespace com.tvd12.ezyfoxserver.client.util
+{
+    public class EzyRoundRobin<T>
+    {
+
+        protected readonly EzyQueue<T> queue;
+
+        public EzyRoundRobin()
+        {
+            this.queue = new EzyQueue<T>();
+        }
+
+        public EzyRoundRobin(Func<T> supplier, int size) : this()
+        {
+            for (int i = 0; i < size; ++i)
+            {
+                this.queue.offer(supplier());
+            }
+        }
+
+        public void add(T item)
+        {
+            lock(queue) {
+                this.queue.add(item);
+            }
+        }
+
+        public T get()
+        {
+            T last = default(T);
+            lock(queue) {
+                last = queue.poll();
+                queue.offer(last);
+            }
```

Comment:
Can we use `queue.peek()` here?

Reply:
Ah nvm, I got it, it takes the first item and adds it back to the end of queue (round-robin)